cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/task/SampleLoadingTask.java

```diff
@@ -4,26 +4,35 @@
 
 package com.linkedin.kafka.cruisecontrol.monitor.task;
 
+import com.linkedin.kafka.cruisecontrol.executor.Executor;
 import com.linkedin.kafka.cruisecontrol.model.ModelParameters;
 import com.linkedin.kafka.cruisecontrol.monitor.sampling.SampleStore;
 import com.linkedin.kafka.cruisecontrol.monitor.sampling.aggregator.KafkaBrokerMetricSampleAggregator;
 import com.linkedin.kafka.cruisecontrol.monitor.sampling.aggregator.KafkaPartitionMetricSampleAggregator;
 
+import static com.linkedin.kafka.cruisecontrol.executor.ExecutorState.State.NO_TASK_IN_PROGRESS;
+
 
 public class SampleLoadingTask implements Runnable {
   private final SampleStore _sampleStore;
   private final KafkaPartitionMetricSampleAggregator _partitionMetricSampleAggregator;
   private final KafkaBrokerMetricSampleAggregator _brokerMetricSampleAggregator;
   private final LoadMonitorTaskRunner _loadMonitorTaskRunner;
+  private final Executor _executor;
 
   SampleLoadingTask(SampleStore sampleStore,
                     KafkaPartitionMetricSampleAggregator partitionMetricSampleAggregator,
                     KafkaBrokerMetricSampleAggregator brokerMetricSampleAggregator,
-                    LoadMonitorTaskRunner loadMonitorTaskRunner) {
+                    LoadMonitorTaskRunner loadMonitorTaskRunner,
+                    Executor executor) {
```

Comment:
Why don't we pass executorState rather than executor to this constructor?

Reply:
This is because currently every time we update the `_executorState` field inside `Executor`, we always doing by assign that field a new  `ExecutorState` object. 

This helps to make `ExecutorState` objects immutable and the good thing about that is the executor state returned in `state` endpoint is always consistent without extra locking.